advanced_alchemy/base.py

```diff
@@ -175,6 +178,41 @@ def to_dict(self, exclude: set[str] | None = None) -> dict[str, Any]:
         return {field.name: getattr(self, field.name) for field in self.__table__.columns if field.name not in exclude}
 
 
+@declarative_mixin
+class SlugKey:
+    """Slug unique Field Model Mixin."""
+
+    @declared_attr
+    def slug(cls) -> Mapped[str]:
+        """Slug field."""
+        return mapped_column(
+            String(length=100),
+            nullable=False,
+        )
+
+    @staticmethod
+    def _create_unique_slug_index(*_args: Any, **kwargs: Any) -> bool:
+        return bool(kwargs["dialect"].name.startswith("spanner"))
+
+    @staticmethod
+    def _create_unique_slug_constraint(*_args: Any, **kwargs: Any) -> bool:
+        return not kwargs["dialect"].name.startswith("spanner")
+
+    @declared_attr.directive
+    def __table_args__(cls) -> tuple:
+        return (
+            UniqueConstraint(
+                cls.slug,
+                name=f"uq_{cls.__tablename__}_slug",  # type: ignore[attr-defined]
+            ).ddl_if(callable_=cls._create_unique_slug_constraint),
+            Index(
+                f"ix_{cls.__tablename__}_slug_unique",  # type: ignore[attr-defined]
+                cls.slug,
+                unique=True,
+            ).ddl_if(callable_=cls._create_unique_slug_index),
+        )
+
+
```

Comment:
This is required for us to maintain Spanner support.  For this dialect, the `UniqueConstraint` is skipped and the `Index` is created instead.